_playwright-tests/UI/SnapshotRepo.spec.ts

```diff
@@ -101,4 +101,132 @@ test.describe('Snapshot Repositories', () => {
       await expect(row).not.toBeVisible();
     });
   });
+
+  test('Snapshot deletion', async ({ page }) => {
+    const smallRHRepo = 'Red Hat CodeReady Linux Builder for RHEL 9 ARM 64 (RPMs)';
+
+    await navigateToRepositories(page);
+    await closePopupsIfExist(page);
+    const repoNamePrefix = 'snapshot-deletion';
+    const randomName = () => `${(Math.random() + 1).toString(36).substring(2, 6)}`;
+    const repoName = `${repoNamePrefix}-${randomName()}`;
+    const templateName = `Test-template-for-snapshot-deletion-${randomName()}`;
+
+    await test.step('Cleanup repositories using "zoo" URLs', async () => {
+      await deleteAllRepos(page, `&search=https://fedorapeople.org/groups/katello/fakerepos/zoo`);
+    });
+
+    await test.step('Create a repository', async () => {
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      await expect(page.getByRole('dialog', { name: 'Add custom repositories' })).toBeVisible();
+      await page.getByLabel('Name').fill(`${repoName}`);
+      await page.getByLabel('Snapshotting').click();
+      await page.getByLabel('URL').fill('https://fedorapeople.org/groups/katello/fakerepos/zoo/');
+      await page.getByRole('button', { name: 'Save', exact: true }).click();
+      const row = await getRowByNameOrUrl(page, repoName);
+      await expect(row.getByText('Valid')).toBeVisible({ timeout: 70000 });
+    });
+
+    await test.step('Edit the repository', async () => {
+      for (let i = 2; i <= 4; i++) {
+        const row = await getRowByNameOrUrl(page, repoName);
+        await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+        await test.step(`Edit repository and create snapshot ${i}`, async () => {
+          // Open the edit modal
+          await row.getByLabel('Kebab toggle').click();
+          await row.getByRole('menuitem', { name: 'Edit' }).click({ timeout: 60000 });
```

Comment:
the dropdown items are located only at the page level in PF6, so to target the Edit menuitem we'll need to change this to `await page.getByRole('menuitem', { name: 'Edit' }).click({ timeout: 60000 });`